### DISCUSSION TITLE
Firebase/Firestore port

### DISCUSSION THREAD

**Discussion by husseyexplores** [kinngh/shopify-node-mongodb-next-app#22] [Show and tell]
Hey `@kinngh`! Thank you so much for this repo. It was a HUGE help. Honestly, can't express enough gratitude 🙏

I needed it to make it work with Firebase/firestore, so I've created a port of this repo here.
https://github.com/husseyexplores/shopify-nextjs-firestore-app

The reason is, I mostly do customizations/custom apps for merchants. I don't build public Shopify apps (yet).
So I find it easier to host the app on Firebase and transfer it to merchants. All in one place.

--- 

Out of curiosity and if you can share - what's your method of sharing access of built apps with a merchant?

Having a separate hosting, database, and possibly any other feature (message queues, social auths, blah blah) and reaching for separate providing for each of them is not manageable if the app is a custom app for one merchant.

By the way, I still use Vercel for hosting. I find Firebase very limiting especially for full-stack apps as they don't allow cookies.

**Comment by kinngh** [comment]
Thank you so much on the kind words and great to see a Fork!

About merchants: I usually have them create a `developer@domain.com` email and then setup whatever accounts I want to and add myself as a collaborator, and setup all apps as a single merchant app distribution and that seems to be going well for me!

**Reply by husseyexplores** [reply]
That's great! 

My question was especially about the payment info stuff.

I usually have them enter their payment info in, say, firebase. And it's all in one place. Billing and services.

If you were to use 3 different services from 3 different SASS providers to build an app - would you have the client enter their payment info in all 3 SASS sites? Or do you do it differently?

**Reply by kinngh** [reply]
It's usually in the lines of "I'm not responsible for your hosting costs and doing my best to optimize my engineering so we don't incur a huge bill, but if we do it's on you". Vercel and Planetscale have really generous free tiers so even with Plus merchants the bills haven't exceeded $60/month.

**Reply by husseyexplores** [reply]
Yup, I do agree. They have extremely generous free tiers. 

And thank you again for this amazing repo! 🚀🚀🚀
